Code/src/main/java/nl/utwente/viskell/ui/components/FunctionBlock.java

```diff
@@ -125,17 +126,21 @@ public final Integer getKnotIndex() {
     }
 
     @Override
-    public Pair<Expression, Set<Block>> getLocalExpr() {
+    public Pair<Expression, Set<OutputAnchor>> getLocalExpr() {
         Expression expr = new FunVar(this.funInfo);
-        Set<Block> addLater = new HashSet<>();
+        Set<OutputAnchor> outsideAnchors = new HashSet<>();
         
         for (InputAnchor in : this.getActiveInputs()) {
-            Pair<Expression, Set<Block>> pair = in.getLocalExpr();
+            Pair<Expression, Set<OutputAnchor>> pair = in.getLocalExpr();
             expr = new Apply(expr, pair.a);
-            addLater.addAll(pair.b);
+            outsideAnchors.addAll(pair.b);
         }
         
-        return new Pair<>(expr, addLater);
+        if (funInfo instanceof DefinitionFunction) {
```

Comment:
Using instanceof here is a bit ugly, what about adding an extra method to FunctionInfo?